CRM/Case/XMLProcessor/Process.php

```diff
@@ -313,28 +313,40 @@ public function activityTypes($activityTypesXML, $maxInst = FALSE, $isLabel = FA
 
   /**
    * @param SimpleXMLElement $caseTypeXML
-   * @return array<string> symbolic activity-type names
+   * @return array[]
+   *   List of activity types along with their component name. Each activity type is an array with keys:
+   *    'name': string; activity type name
+   *    'component_name': string; activity type component name ( e.g : CiviCase )
    */
   public function getDeclaredActivityTypes($caseTypeXML) {
     $result = array();
 
+    $componentName = 'CiviCase';
+    if (!empty($caseTypeXML->componentName)) {
+      $componentName = (string) $caseTypeXML->componentName;
+    }
+
     if (!empty($caseTypeXML->ActivityTypes) && $caseTypeXML->ActivityTypes->ActivityType) {
       foreach ($caseTypeXML->ActivityTypes->ActivityType as $activityTypeXML) {
-        $result[] = (string) $activityTypeXML->name;
+        $result[] = array(
+          'name' => (string) $activityTypeXML->name,
+          'component_name' => $componentName);
       }
     }
 
     if (!empty($caseTypeXML->ActivitySets) && $caseTypeXML->ActivitySets->ActivitySet) {
       foreach ($caseTypeXML->ActivitySets->ActivitySet as $activitySetXML) {
         if ($activitySetXML->ActivityTypes && $activitySetXML->ActivityTypes->ActivityType) {
           foreach ($activitySetXML->ActivityTypes->ActivityType as $activityTypeXML) {
-            $result[] = (string) $activityTypeXML->name;
+            $result[] = array(
+              'name' => (string) $activityTypeXML->name,
+              'component_name' => $componentName);
```

Comment:
Closing array parenthesis should go on the next line (also see L333 for the same issue).